src/main/scala/mesosphere/marathon/core/groupmanager/impl/GroupManagerActor.scala

```diff
@@ -0,0 +1,279 @@
+package mesosphere.marathon.core.groupmanager.impl
+
+import java.net.URL
+import javax.inject.Provider
+
+import akka.actor.{ Actor, ActorLogging, Props }
+import akka.event.EventStream
+import akka.pattern.pipe
+import mesosphere.marathon._
+import mesosphere.marathon.api.v2.Validation._
+import mesosphere.marathon.core.event.{ GroupChangeFailed, GroupChangeSuccess }
+import mesosphere.marathon.core.task.Task
+import mesosphere.marathon.io.PathFun
+import mesosphere.marathon.io.storage.StorageProvider
+import mesosphere.marathon.state.{ AppDefinition, Container, PortDefinition, _ }
+import mesosphere.marathon.upgrade.{ DeploymentPlan, GroupVersioningUtil, ResolveArtifacts }
+import mesosphere.util.CapConcurrentExecutions
+import org.slf4j.LoggerFactory
+
+import scala.collection.immutable.Seq
+import scala.collection.mutable
+import scala.concurrent.Future
+import scala.util.{ Failure, Success }
+
+private[groupmanager] object GroupManagerActor {
```

Comment:
@jasongilanfarr this is only the first step. With an actor in place, we can hold the rootGroup, dependencyTree etc. Then the actor will make more sense.

@nfnt Can you add the `Response` types and document which request will send which response?